assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.*;
+
+import static java.util.Map.Entry.comparingByValue;
+import static java.util.stream.Collectors.toMap;
+
+
+public class IPL {
+
+    static String[] teamNames = {"CSK", "MI", "RR", "RCB", "SRH", "PBKS", "DC", "KKR"};
+    static String[] venueNames = {"CHENNAI", "MUMBAI", "JAIPUR", "BENGALURU", "HYDERABAD", "CHANDIGARH", "DELHI", "KOLKATA"};
+    static ArrayList<Team> teams = new ArrayList<>();
+    static ArrayList<Match> matches = new ArrayList<>();
+
+    public static void fortyWicketsClub(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
```

Comment:
You can have a map storing team names and the team object as a key-value pair.

Reply:
Okay, actually I had done using map only, but then I thought teams size will be 8 or 100 in very worst case practically, so I thought I should try linear filtering.